### DISCUSSION TITLE
Problems in calculating internal resistance

### DISCUSSION THREAD

**Discussion by MarcoZangrandi** [pybamm-team/liionpack#305] [Q&A]
Hi, I modified the script found [here](https://github.com/pybamm-team/liionpack/blob/develop/examples/different_initial_soc.py) because copied and pasted it gives this error ‘ValueError: Initial value must be a float between 0 and 1, or a string ending in “V”’. 
The script I attach below simulates a parallel of three cells with three different initial SOCs (100%, 50% and 80%). The results of the simulation, however, are a bit strange. 

In particular, the internal resistance of cell 2 has a strange trend as cell 2 goes from a charging state to a discharging state (i.e. it goes for cell currents very close to 0). How could this problem be solved? I don't think there is any physical significance to this sudden change in resistance.

```
"""
Example of running a simulation with three batteries of different initial SOC.
"""

import liionpack as lp
import pybamm
import numpy as np
import matplotlib.pyplot as plt

lp.logger.setLevel("NOTICE")

# Define parameters
Np = 3
Ns = 1
Nbatt = Np * Ns
Iapp = 5
SOC_init = [1, 0.5, 0.8]
conc_n_init = []
conc_p_init = []

# Generate the netlist and output variables
netlist = lp.setup_circuit(Np=Np, Ns=Ns, Rb=1.5e-6, Rc=1e-6, Ri=5e-2, V=4.0, I=Iapp)
output_variables = [
    "X-averaged negative particle surface concentration [mol.m-3]",
    "X-averaged positive particle surface concentration [mol.m-3]",
    "Battery open-circuit voltage [V]"
]

# Cycling experiment
experiment = pybamm.Experiment(
    [
        f"Discharge at {Iapp} A for 3 hours",
    ],
    period="5 seconds",
)

# PyBaMM parameters
param = pybamm.ParameterValues("Chen2020")

for index in range(Nbatt):
    c_s_n_init, c_s_p_init = lp.update_init_conc(
        param, SoC=SOC_init[index], update=False
        )
    conc_n_init.append(c_s_n_init)
    conc_p_init.append(c_s_p_init)


param.update(
    {
        "Initial concentration in negative electrode [mol.m-3]": "[input]",
        "Initial concentration in positive electrode [mol.m-3]": "[input]",
    }
)

inputs = {
    "Initial concentration in negative electrode [mol.m-3]": conc_n_init,
    "Initial concentration in positive electrode [mol.m-3]": conc_p_init,
}

def thermal_sim(parameter_values=None):
    """
    Create a PyBaMM simulation set up for integration with liionpack

    Args:
        parameter_values (pybamm.ParameterValues):
            The default is None.

    Returns:
        pybamm.Simulation:
            A simulation that can be solved individually or passed into the
            liionpack solve method

    """
    # Create the pybamm model
    model = pybamm.lithium_ion.SPMe(
        options={
            "thermal": "lumped",
        }
    )

    # Add events to the model
    model = lp.add_events_to_model(model)

    # Set up parameter values
    if parameter_values is None:
        parameter_values = pybamm.ParameterValues("Chen2020")

    # Set up solver and simulation
    solver = pybamm.CasadiSolver(mode="safe")
    sim = pybamm.Simulation(
        model=model,
        parameter_values=parameter_values,
        solver=solver,
    )
    return sim

# Solve pack
output = lp.solve(
    netlist=netlist,
    sim_func=thermal_sim,
    parameter_values=param,
    experiment=experiment,
    output_variables=output_variables,
    initial_soc=None,
    inputs=inputs,
)

# Plot results
lp.plot_output(output)
lp.show_plots()
```
![internal_resistances](https://github.com/pybamm-team/liionpack/assets/167761111/cba89521-915d-485f-9492-a85bcbba34da)
![currents](https://github.com/pybamm-team/liionpack/assets/167761111/f2e13370-3481-40d7-ba55-72e4211b7ac4)
![surf_ocv](https://github.com/pybamm-team/liionpack/assets/167761111/d74a9593-3a8d-4caf-800d-86609afbdb14)
![term_v](https://github.com/pybamm-team/liionpack/assets/167761111/c759fbac-248f-4dc0-b04c-244eae2fa577)

**Comment by MarcoZangrandi** [comment]
Does anyone have any news on this?

**Reply by TomTranter** [reply]
Yeah it does look a bit strange but the current profile looks sensible. The resistance is very sensitive to small currents as it's just getting calculated by ohm's law. It could probably be improved but doesn't look to be affecting the current or voltage in a bad way.

**Reply by MarcoZangrandi** [reply]
Hi Tom, thank you for your reply. Would you have any ideas on how this aspect of resistance could be improved?

**Reply by TomTranter** [reply]
Moving the solving to PyBaMM is on the long term road map which will have better time stepping
